File: fleytitala.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

double d; ll k;
int main() {
    scanf("%lf\n%lld",&d,&k);
    double ans = d;
    // after 17 iterations, desired accuracy is attained
    for (ll i = 0; i < min(17ll, k); i++) {
        d *= 0.5;
        ans += d;
    }
    printf("%lf\n",ans);
    return 0;
}

File: fleytitala.java
import java.io.*;

public class fleytitala {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
        double d = Double.parseDouble(br.readLine());
        long k = Long.parseLong(br.readLine());
        double ans = d;
        // after 17 iterations, desired accuracy is attained
        for (long i = 0; i < Math.min(17, k); i++) {
            d *= 0.5;
            ans += d;
        }
        pw.println(ans);
        br.close();
        pw.close();
    }
}
